Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+                {
+                    WasBraking = false;
+                    if (SpeedRegMode == SpeedRegulatorMode.Auto && UseThrottleAsForceSelector) Locomotive.ThrottleController.SetPercent(0);
+                    Locomotive.SetThrottlePercent(0);
+                }
+                if (ForceResetRequiredAfterBraking && WasBraking && (SelectedMaxAccelerationStep > 0 || SelectedMaxAccelerationPercent > 0))
+                {
+                    Locomotive.SetThrottlePercent(0);
+                    controllerVolts = 0;
+                    maxForceN = 0;
+                    if (SpeedSelMode == SpeedSelectorMode.Parking)
+                        if (AbsWheelSpeedMpS < (SpeedIsMph ? MpS.FromMpH(ParkingBrakeEngageSpeed) : MpS.FromKpH(ParkingBrakeEngageSpeed)))
+                            Locomotive.SetEngineBrakePercent(ParkingBrakePercent);
+                    return;
+                }
+
+                if (ForceResetRequiredAfterBraking && !WasForceReset)
+                {
+                    Locomotive.SetThrottlePercent(0);
+                    controllerVolts = 0;
+                    maxForceN = 0;
+                    if (SpeedSelMode == SpeedSelectorMode.Parking)
+                        if (AbsWheelSpeedMpS < (SpeedIsMph ? MpS.FromMpH(ParkingBrakeEngageSpeed) : MpS.FromKpH(ParkingBrakeEngageSpeed)))
+                            Locomotive.SetEngineBrakePercent(ParkingBrakePercent);
+                    return;
+                }
+
+
+                if (canAddForce)
+                {
+                    if (Locomotive.AbsSpeedMpS == 0)
+                    {
+                        timeFromEngineMoved = 0;
+                        reducingForce = true;
+                    }
+                    else if (reducingForce)
+                    {
+
+                        timeFromEngineMoved += elapsedClockSeconds;
+                        float timeToReduce = Locomotive.SelectedTrainType == MSTSLocomotive.TrainType.Pax ? PowerReductionDelayPaxTrain : PowerReductionDelayCargoTrain;
+                        if (timeFromEngineMoved > timeToReduce)
+                            reducingForce = false;
+                    }
+                }
+                if (UseTrainBrakeAndDynBrake && CCIsUsingTrainBrake)
+                {
+                    canAddForce = true;
+                }
+                else if (Bar.FromPSI(Locomotive.BrakeSystem.BrakeLine1PressurePSI) < 4.98)
+                {
+                    canAddForce = false;
+                    reducingForce = true;
+                    timeFromEngineMoved = 0;
+                    maxForceN = 0;
+                    if (controllerVolts > 0)
+                        controllerVolts = 0;
+                    Ampers = 0;
+                    Locomotive.ThrottleController.SetPercent(0);
+                    return;
+                }
+                else if (Bar.FromPSI(Locomotive.BrakeSystem.BrakeLine1PressurePSI) > 4.7)
```

Comment:
Only for UIC brakes.

Reply:
As above

Reply:
MaxSystemPressure - MinReductionBar - 0.1bar

Reply:
In my opinion the if condition of this else is always satisfied in the original code, because the code arrives here only if Bar.FromPSI(Locomotive.BrakeSystem.BrakeLine1PressurePSI) is higher than 4.98 (preceding else if). What do you think?

Reply:
Yes, you are right. Then I think the code could be changed to:
```csharp
if ((UseTrainBrakeAndDynBrake && CCIsUsingTrainBrake) || Locomotive.BrakeSystem.BrakeLine1PressurePSI > TrainBrakeController.MaxPressurePSI - 1){}
else{}
```
(not sure about the proper naming of TrainBrakeController.MaxPressurePSI)

Reply:
Done, but with an AND and not an OR.